rust/plugin-lib/src/lib.rs

```diff
@@ -109,6 +110,11 @@ pub trait Cache {
 pub trait Plugin {
     type Cache: Cache;
 
+    /// Called when the Plugin is initialized. RpcCtx is supplied to
+    /// the plugin so that it can stash a reference to peer in case it 
+    /// wants to send async notifications to the core later
+    fn initialize(&mut self, ctx: &RpcCtx);
```

Comment:
You can include a default implementation in the trait itself, which means plugins aren't required to implement it at all.